docs/diagrams/wallet-lifecycle/wallet-lifecycle-1-handling-frauds.tikz

```diff
@@ -0,0 +1,187 @@
+
+% Arrows
+\draw[->] (fc_start)   -- (fc_check1);
+\draw[->] (fc_check1)  -| (fc_check2)  \no;
+\draw[->] (fc_check1)  -| (fc_revert1) \yes;
+\draw[->] (fc_check2)  -| (fc_revert2) \no;
+\draw[->] (fc_check2)  -| (fc_check3)  \yes;
+\draw[->] (fc_check3)  -| (fc_revert2) \no;
+\draw[->] (fc_check3)  -| (fc_action1) \yes;
+
+
+% DEFEAT FRAUD
+% Nodes
+\node[start, right=of fc_start, xshift=100mm]                    (df_start)   {\textbf{Defeat fraud challenge}};
+\node[check, below=of df_start]                                  (df_check1)  {Is there a pending\\ notification to be defeated?};
+\node[check, below right=of df_check1]                           (df_check2)  {Has the defeat timeout\\ been reported?};
+\node[check, below left=of df_check2]                            (df_check3)  {Were the UTXOs\\ in the challenged transaction\\ spent non-fraudulently?};
+\node[revert, below right=of df_check2]                          (df_revert1) {Transaction reverted};
+\node[revert, below left=of df_check3]                           (df_revert2) {Transaction reverted};
+\node[action, below right=of df_check3, yshift=-3cm]             (df_action1) {Close the fraud challenge, don't allow for challenging again)};
+\node[action, below=of df_action1]                               (df_action2) {Transfer the challenger's deposit to the treasury};
+\node[note1, right=of df_start.south east, anchor=south west]    (df_note1)   {Can be successfully called for wallets:\\ 
+                                                                               \textbf{- LIVE\\
+                                                                               - MOVING\_FUNDS\\
+                                                                               - CLOSING}};
+\node[note2, above=of df_note1.north, anchor=south, yshift=-8mm] (df_note2)   {Sender incentivized by:\\ \textbf{Receiving recompensation from the treasury}};
+
+% Arrows
+\draw[->] (df_start)   -- (df_check1);
+\draw[->] (df_check1)  -| (df_revert2) \no;
+\draw[->] (df_check1)  -| (df_check2)  \yes;
+\draw[->] (df_check2)  -| (df_check3)  \no;
+\draw[->] (df_check2)  -| (df_revert1) \yes;
+\draw[->] (df_check3)  -| (df_revert2) \no;
+\draw[->] (df_check3)  -| (df_action1) \yes;
+\draw[->] (df_action1) -- (df_action2);
+
+
+% DEFEAT FRAUD CHALLENGE WITH HEARTBEAT
+% Nodes
+\node[start, right=of df_start, xshift=110mm]                      (dfwt_start)   {\textbf{Defeat fraud challenge with heartbeat}};
+\node[check, below=of dfwt_start]                                  (dfwt_check1)  {Is there a pending\\ notification to be defeated?};
+\node[check, below right=of dfwt_check1]                           (dfwt_check2)  {Has the defeat timeout\\ been reported?};
+\node[check, below left=of dfwt_check2]                            (dfwt_check3)  {Is the\\ attatched message\\ a valid heartbeat\\ message?};
```

Comment:
I would replace "attached" with "provided" given the heartbeat message is a parameter to the function and we are usually *providing* function parameters.